inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/config/filebased/ConfigurationDirectoriesPoller.java

```diff
@@ -0,0 +1,148 @@
+import rocks.inspectit.oce.core.config.model.config.FileBasedConfigSettings;
+import rocks.inspectit.oce.core.service.DynamicallyActivatableService;
+
+import java.io.File;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Service for automatically polling all registered {@link DirectoryPropertySource}s.
+ * If any changes are detected, the configuration gets reloaded into the {@link InspectitEnvironment}.
+ */
+@Service
+@Slf4j
+public class ConfigurationDirectoriesPoller extends DynamicallyActivatableService {
+
+    @Autowired
+    InspectitEnvironment env;
+
+    @Autowired
+    ScheduledExecutorService executor;
+
+    private ScheduledFuture<?> pollingTask = null;
+
+    private List<DirectoryPropertySourceChangeChecker> filePollers;
+
+    public ConfigurationDirectoriesPoller() {
+        super("config.fileBased");
+    }
+
+    @Override
+    protected boolean checkEnabledForConfig(InspectitConfig conf) {
+        FileBasedConfigSettings fileBased = conf.getConfig().getFileBased();
+        return fileBased.isWatch() && fileBased.getFrequency().toMillis() > 0;
+    }
+
+    @Override
+    protected boolean doEnable(InspectitConfig conf) {
+        log.info("Starting config directory polling service..");
+        env.readPropertySources(propertySources ->
+                filePollers = propertySources.stream()
+                        .filter(ps -> ps instanceof DirectoryPropertySource)
+                        .map(ps -> (DirectoryPropertySource) ps)
+                        .map(ps -> createCheckerForDirectory(ps))
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .collect(Collectors.toList())
+        );
+        if (filePollers.isEmpty()) {
+            log.info("No directories to watch registered, terminating poll service.");
+            return false;
+        } else {
+            long freqMS = conf.getConfig().getFileBased().getFrequency().toMillis();
+            pollingTask = executor.scheduleWithFixedDelay(this::pollDirectories, freqMS, freqMS, TimeUnit.MILLISECONDS);
+            return true;
+        }
+    }
+
+    private void pollDirectories() {
+        for (val poller : filePollers) {
+            poller.checkForChangesAndReloadIfRequired();
+        }
+    }
+
+
+    @Override
+    protected boolean doDisable() {
+        if (pollingTask != null) {
+            log.info("Stopping config directory polling service.");
+            pollingTask.cancel(true);
+            for (val poller : filePollers) {
+                try {
+                    poller.destroy();
+                } catch (Exception e) {
+                    log.error("Error destroying poller!", e);
+                }
+            }
+        }
+        return true;
```

Comment:
clear file poolers list, when activated again you would have duplicated entries? ahh I see you will create a new list.. anyway, let's ensure this one is empty..